src/Angular.js

```diff
@@ -1454,8 +1454,7 @@ function allowAutoBootstrap(document) {
   var link = document.createElement('a');
   link.href = src;
   var scriptProtocol = link.protocol;
-  var docLoadProtocol = document.location.protocol;
-  if (docLoadProtocol === scriptProtocol) {
+  if (document.location.origin === link.origin) {
```

Comment:
Ah I get it, you're talking about extensions protecting themselves using CSP. That's a fair point, this code wasn't intended to cover this but we might as well.

* can you add a comment explaining that this covers two cases, loading Angular from an extension into a website, and across extensions?
* can you add a test that demonstrates this, so we don't break it in the future?

Also a nit: can you move the `var scriptProtocol` and inline it into the switch below? We no longer need the separate variable here.